src/ray/core_worker/core_worker.cc

```diff
@@ -415,6 +421,13 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
                           : std::shared_ptr<LeasePolicyInterface>(
                                 std::make_shared<LocalLeasePolicy>(rpc_address_));
 
+  std::function<uint64_t()> get_max_pending_lease_requests_per_scheduling_category =
```

Comment:
Why create this as a lambda? Seems like it should be a function on the class to avoid bloating this function?

This is another +1 to creating a small class to track the `num_alive_nodes` state; it can have `increment`/`decrement` functions (with appropriate logs and asserts) and a `get` that does this config check. Will improve testability as well, can unit test just this class!